linera-execution/src/execution_state_actor.rs

```diff
@@ -336,8 +336,24 @@ where
                     .map(|http::Header { name, value }| Ok((name.parse()?, value.try_into()?)))
                     .collect::<Result<HeaderMap, ExecutionError>>()?;
 
+                let url = Url::parse(&request.url)?;
+                let host = url
+                    .host_str()
+                    .ok_or_else(|| ExecutionError::UnauthorizedHttpRequest(url.clone()))?;
```

Comment:
Did you mean to use `InvalidUrlForHttpRequest` here instead?

Reply:
Not really, honestly I think either is fine :thinking: . However the `UnauthorizedHttpRequest` provides the URL in the error :man_shrugging: